src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLanguageServerClient.cs

```diff
@@ -284,11 +287,27 @@ private void ServerStarted()
     {
         _projectConfigurationFilePathStore.Changed += ProjectConfigurationFilePathStore_Changed;
 
+        if (_languageServerFeatureOptions.DoNotUseProjectConfigurationFile)
+        {
+            _projectInfoEndpointPublisher.StopCachingRequests();
+        }
+
         var mappings = _projectConfigurationFilePathStore.GetMappings();
         foreach (var mapping in mappings)
         {
             var args = new ProjectConfigurationFilePathChangedEventArgs(mapping.Key, mapping.Value);
             ProjectConfigurationFilePathStore_Changed(this, args);
+
+            if (_languageServerFeatureOptions.DoNotUseProjectConfigurationFile)
+            {
+                _projectInfoEndpointPublisher.SerializeToEndpointUncached(mapping.Key, mapping.Value);
+            }
+        }
+
+
+        if (_languageServerFeatureOptions.DoNotUseProjectConfigurationFile)
+        {
+            _projectInfoEndpointPublisher.ClearCache();
```

Comment:
This another part of the publisher that feels odd, having its cache, which feels like it should be an implementation detail, be managed by an outside service.

Reply:
Agreed, mostly because I was takign "replace file writing with request sending" too literally. I like these suggestions, working on them now.

Reply:
No longer need cache after following your suggestion, thanks!